src/executionengine/lljit.jl

```diff
@@ -0,0 +1,82 @@
+@checked struct LLJITBuilder
+    ref::API.LLVMOrcLLJITBuilderRef
+end
+Base.unsafe_convert(::Type{API.LLVMOrcLLJITBuilderRef}, builder::LLJITBuilder) = builder.ref
+
+@checked mutable struct LLJIT
+    ref::API.LLVMOrcLLJITRef
+end
+Base.unsafe_convert(::Type{API.LLVMOrcLLJITRef}, lljit::LLJIT) = lljit.ref
+
+function LLJITBuilder()
+    ref = API.LLVMOrcCreateLLJITBuilder()
+    LLJITBuilder(ref)
+end
+
+function dispose(builder::LLJITBuilder)
+    API.LLVMOrcDisposeLLJITBuilder(builder)
+end
+
+function targetmachinebuilder!(builder::LLJITBuilder, tmb::TargetMachineBuilder)
+    API.LLVMOrcLLJITBuilderSetJITTargetMachineBuilder(builder, tmb)
+end
+
+function linkinglayercreator!(builder::LLJITBuilder, callback, ctx)
+    API.LLVMOrcLLJITBuilderSetObjectLinkingLayerCreator(builder, callback, ctx)
+end
+
+"""
+    LLJIT(::LLJITBuilder)
+
+Creates a LLJIT stack based on the provided builder.
+
+!!! note
+    Takes ownership of the provided builder.
+"""
+function LLJIT(builder::LLJITBuilder)
+    ref = Ref{API.LLVMOrcLLJITRef}()
+    @check API.LLVMOrcCreateLLJIT(ref, builder)
+    LLJIT(ref[])
+end
+
+function LLJIT(; tm::Union{Nothing, TargetMachine} = nothing)
```

Comment:
Why a kwarg? I'd make this positional.

Reply:
The idea was to eventually expose other LLJITBuilder interfaces here.  Like the objectlayerbuilder.

Reply:
Ah, OK. Not sure it's worth an additional constructor, but we can keep it like that for now.